packages/x-date-pickers/src/MultiSectionDigitalClock/MultiSectionDigitalClock.types.ts

```diff
@@ -1,3 +1,4 @@
+import * as React from 'react';
```

Comment:
We are using `React` inside this page without importing it.
My IDE complains if we don't import it, based on [this blog post](https://www.totaltypescript.com/react-refers-to-a-umd-global) maybe it's because we have some `"jsx": "react"` in some tsconfig that the IDE is using.

Since we are importing it most of the time, I'm adding it whenever I see one missing to remove this error.


image](https://github.com/mui/mui-x/assets/3309670/eb36a5e5-e631-44c9-883f-c549ac1ee49d)

Reply:
Fair point, didn't see that entry. 🙈 
VSCode doesn't complain... 🤷